x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java

```diff
@@ -70,8 +96,25 @@ public ActionRequestValidationException validate() {
                 validationException = addValidationError("[connector_id] cannot be [null] or [\"\"].", validationException);
             }
 
+            // If [filtering] is not present in the request payload it means that the user should define both [rules] and [advances_snippet]
             if (filtering == null) {
-                validationException = addValidationError("[filtering] cannot be [null].", validationException);
+                if (rules == null && advancedSnippet == null) {
+                    validationException = addValidationError(
+                        "[filtering], [advanced_snippet] and [rules] cannot be all [null].",
+                        validationException
+                    );
+                } else if (rules == null) {
+                    validationException = addValidationError("[rules] cannot be [null].", validationException);
+                } else if (advancedSnippet == null) {
+                    validationException = addValidationError("[advanced_snippet] cannot be [null].", validationException);
+                }
+            }
+            // If [filtering] is present we don't expect [rules] and [advances_snippet] in the request body
+            else if (rules != null || advancedSnippet != null) {
+                validationException = addValidationError(
+                    "If [filtering] is specified, [rules] and [advanced_snippet] should not be present in the request body.",
+                    validationException
+                );
```

Comment:
Should we add another check indicating that the `domain` cannot be updated?

Reply:
Added core logic to prevent users from updating `domain`